crafters/image/ImageReader/__init__.py

```diff
@@ -39,4 +39,23 @@ def craft(self, buffer: bytes, uri: str, *args, **kwargs) -> Dict:
         img = np.array(raw_img).astype('float32')
         if self.channel_axis != -1:
             img = np.moveaxis(img, -1, self.channel_axis)
-        return dict(weight=1., blob=img)
+        # Add image metadata
+        from PIL.ExifTags import TAGS
+        from PIL.ExifTags import GPSTAGS 
+        # Extract image metadata
+        # https://github.com/python-pillow/Pillow/blob/master/src/PIL/ExifTags.py
+        metadata = {}
+        exif_data = raw_img.getexif()
+        for tag_id in exif_data:
+            tag = TAGS.get(tag_id)
+            if tag is None:
+                tag = GPSTAGS.get(tag_id, tag_id)
```

Comment:
what is this logic here? why passing tag_id twice?

Reply:
From [Pillow Code](https://github.com/python-pillow/Pillow/blob/master/src/PIL/ExifTags.py) we have two dictionaries (TAGS and GPSTAGS), which transform each tag ID into a human-readable string. The TAGS dictionary has tags that are general characteristics (features). GPSTAGS are specifically related to GPS locations.